Doc and release notes followup on #26715 (#27431)
zephyrproject-rtos/zephyr: doc/releases/release-notes-2.4.rst

* Devicetree

> * Support of multiple SOC and ARCH roots in Zephyr.
>   In order to support ``SOC_ROOT`` to support multiple paths, then it has been
>   necessary to deprecate the ``SOC_DIR`` variable in Kconfig.
>   Usage of ``source $(SOC_DIR)/<path>`` must should be adjusted into
>   ``rsource <relative>/<path>`` or similar.

Reviewer: ```suggestion
* Support for multiple SOC and ARCH roots.
  The :ref:`SOC_ROOT <application>` and ``ARCH_ROOT`` variables used to
  specify support files for out of tree SoCs and architectures now accept multiple paths,
  separated by semicolons. As a result, the ``SOC_DIR`` Kconfig variable is no longer
  supported.
  
  Uses like ``source $(SOC_DIR)/<path>`` must be changed to
  ``rsource <relative>/<path>`` or similar.
```

> * BOARD, SOC, DTS, and ARCH roots can now be specified in the Zephyr modules
>   :file:`zephyr/module.yml` file.

Reviewer: ```suggestion
* BOARD, SOC, DTS, and ARCH roots can now be specified in each module's
  :file:`zephyr/module.yml` file; see :ref:`modules_build_settings`.
```
